Author: Pino Toscano <pino@debian.org>
Description: Do not require executable permissions on plugins.
 There is no need for plugins to have executable permissions, as they work fine
 without them.  Also, libraries have these permission bits stripped in Debian
 systems, breaking the plugin loading.
 .
 No need to forward this upstream, since there is a KPluginLoader-based system
 there now.
Last-Update: 2016-03-25
Forwarded: not-needed

--- a/libkwave/PluginManager.cpp
+++ b/libkwave/PluginManager.cpp
@@ -632,7 +632,7 @@ void Kwave::PluginManager::searchPluginM
 	QDir d(dir);
 	qDebug("searching for plugins in '%s'", DBG(dir));
 	QStringList f = d.entryList(
-	    QDir::Files | QDir::Readable | QDir::Executable, QDir::Name);
+	    QDir::Files | QDir::Readable, QDir::Name);
 	foreach (const QString &file, f) {
 	    if (file.startsWith(_("kwaveplugin_")))
 		files += d.path() + sep + file;
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -65,9 +65,6 @@ MACRO(KWAVE_PLUGIN _plugin)
         plugin_${_plugin}
         LIBRARY
         DESTINATION ${PLUGIN_INSTALL_DIR}
-        PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
-                    GROUP_READ GROUP_EXECUTE
-                    WORLD_READ WORLD_EXECUTE
     )
 
 ENDMACRO(KWAVE_PLUGIN)
